pkg/cli/cli.go

```diff
@@ -90,14 +91,26 @@ func Run(configPath string, inputPath string, outputPath string, deleteOutputFol
 
 	traversal.NewParallelFileWalker(inputPath, workerCount, workerFactory).Traverse()
 
-	reporter := reporting.NewSimpleReporter()
-	reporter.CollectOmitterReport(mro.Report())
-	reporter.CollectObfuscatorReport(mo.ReportPerObfuscator())
-	reporterErr := reporter.WriteReport(filepath.Join(reportingFolder, reportFileName))
-	if reporterErr != nil {
-		return reporterErr
+	mo.UpdateReportPerObfuscator(config)
+	data, err := yaml.Marshal(config)
+	if err != nil {
+		return fmt.Errorf("failed to encode the report at %s: %w", reportFileName, err)
 	}
 
+	err = ioutil.WriteFile(filepath.Join(reportingFolder, reportFileName), data, 0644)
+	if err != nil {
+		return fmt.Errorf("failed to generate the report at %s: %w", reportFileName, err)
+	}
+
+	/*
+		reporter := reporting.NewSimpleReporter()
+		reporter.CollectOmitterReport(mro.Report())
+		reporter.CollectObfuscatorReport(mo.ReportPerObfuscator())
+		reporterErr := reporter.WriteReport(filepath.Join(reportingFolder, reportFileName))
+		if reporterErr != nil {
+			return reporterErr
+		}
+	*/
```

Comment:
I love that you remove code, I think we still need it (very least for the omissions) and the business logic in `UpdateReportPerObfuscator` should go into this reporting package. I'd extend this by adding a config to the reporter and then it can hash out what to merge with which result.